platform/ios/src/MGLScaleBarView.m

```diff
@@ -0,0 +1,387 @@
+    if (_metersPerPoint == metersPerPoint) {
+        return;
+    }
+    
+    [self fadeIn];
+    
+    _metersPerPoint = metersPerPoint;
+    self.row = [self preferredRowInRows:[self validRows]];
+    
+    [self invalidateIntrinsicContentSize];
+    [self setNeedsLayout];
+}
+
+- (void)setRow:(NSArray<NSNumber *> *)row {
+    if (![row isEqualToArray:_row]) {
+        [_bars makeObjectsPerformSelector:@selector(removeFromSuperview)];
+        [_labels makeObjectsPerformSelector:@selector(removeFromSuperview)];
+        _bars = nil;
+        _labels = nil;
+        _row = row;
+    }
+}
+
+#pragma mark - Views
+
+- (NSArray<UIView *> *)bars {
+    if (!_bars) {
+        NSMutableArray *bars = [NSMutableArray array];
+        for (NSUInteger i = 0; i < self.row.lastObject.integerValue; i++) {
+            UIView *bar = [[UIView alloc] init];
+            [bars addObject:bar];
+            [self.containerView addSubview:bar];
+        }
+        _bars = bars;
+    }
+    return _bars;
+}
+
+- (NSArray<UILabel *> *)labels {
+    if (!_labels) {
+        NSMutableArray *labels = [NSMutableArray array];
+        for (NSUInteger i = 0; i <= self.row.lastObject.integerValue; i++) {
+            UILabel *label = [[MGLScaleBarLabel alloc] init];
+            label.font = [UIFont systemFontOfSize:8 weight:UIFontWeightMedium];
+            NSCharacterSet *characterSet = [[NSCharacterSet decimalDigitCharacterSet] invertedSet];
+            label.text = [[[self.formatter stringFromDistance:0] componentsSeparatedByCharactersInSet:characterSet] componentsJoinedByString:@""];
+            [label setNeedsDisplay];
+            [label sizeToFit];
+            [labels addObject:label];
+            [self addSubview:label];
+        }
+        _labels = labels;
+    }
+    return _labels;
+}
+
+#pragma mark - Layout
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+    
+    if (!self.row.lastObject || !self.row.lastObject.integerValue) {
+        // Current distance is not within allowed range
+        return;
+    }
+    
+    [self updateLabels];
+    [self layoutBars];
+    [self layoutLabels];
+}
+
+- (void)layoutBars {
+    CGFloat barWidth = (CGRectGetWidth(self.bounds) - self.borderWidth * 2.0f) / self.bars.count;
+    
+    NSUInteger i = 0;
+    for (UIView *bar in self.bars) {
+        CGFloat xPosition = barWidth * i + self.borderWidth;
+        bar.backgroundColor = (i % 2 == 0) ? self.primaryColor : self.secondaryColor;
+        bar.frame = CGRectMake(xPosition, self.borderWidth, barWidth, MGLBarHeight);
+        i++;
+    }
+    
+    self.containerView.frame = CGRectMake(CGRectGetMinX(self.bars.firstObject.frame),
+                                          CGRectGetMaxY(self.bounds)-MGLBarHeight,
+                                          self.actualWidth,
+                                          MGLBarHeight+self.borderWidth*2);
+    
+    self.containerView.layer.cornerRadius = 2.0f;
```

Comment:
Any objections to getting rid of the corner radius here? (The white border around the outside.)

<img width="201" alt="screen shot 2017-04-05 at 12 11 52 am" src="https://cloud.githubusercontent.com/assets/1198851/24689432/7fa3fb22-1994-11e7-99c8-31d4a2862b85.png">

It gives me a vague sense of unevenness, probably because the corners of the black outline are not rounded.